extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxCurrentContextFactory.java

```diff
@@ -41,20 +54,52 @@ public <T extends InjectableContext.ContextState> CurrentContext<T> create(Class
         return new VertxCurrentContext<>(key);
     }
 
-    /**
-     *
-     * @return an unmodifiable list of used keys
-     */
-    public List<String> keys() {
-        return unmodifiableKeys;
+    public ContextInternal duplicateContextIfContainsAnyCreatedScopeKeys(ContextInternal vertxContext) {
+        if (!containsAnyCreatedScopeKeys(vertxContext)) {
+            return vertxContext;
+        }
+        // Duplicate the context, copy the data, remove the request context
+        var duplicateCtx = vertxContext.duplicate();
+        // TODO this is not copying any ContextLocal<?> from the original context to the new one!
```

Comment:
@vietj @mariofusco This is just not right :"( : in short, given that there is no `ContextInternal`'s API to retrieve the existing `ContextLocal` keys (apart from the `ConcurrentHashMap` at index 0!!!!), 
IDK which ones should be copied and which not....

Having a `ContextInternal::putLocal(AccessMode mode, ContextInternal from)` method would be great here, because can skip any collection allocation e.g. `ContextInternal::getContextLocals` would force some allocation to happen.

The same (even better) could be achieved with `ContextInternal::duplicate(bool copyLocalStorage)` which can further avoid allocating any `ConcurrentHashMap` (happening at `ContextInternal::localContextData` call, regadless we have nothing in :"( )

wdyt @mariofusco do you want to send such PR to vertx? 
@vietj do you agree on this?